PurchasesCoreSwift/FoundationExtensions/DateFormatter+Extensions.swift

```diff
@@ -0,0 +1,31 @@
+//
+// Created by Andrés Boedo on 7/29/20.
+// Copyright (c) 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+extension DateFormatter {
+    static let iso8601SecondsDateFormatter: DateFormatter = {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
+        return dateFormatter
+    }()
+
+    static let iso8601MilliSecondsDateFormatter: DateFormatter = {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+        return dateFormatter
+    }()
+
+    func iso8601SecondsOrMillisecondsDate(fromBytes bytes: ArraySlice<UInt8>) -> Date? {
+        guard let dateString = String(bytes: Array(bytes), encoding: .ascii) else { return nil }
+        return (Self.iso8601SecondsDateFormatter.date(from: dateString)
+            ?? Self.iso8601MilliSecondsDateFormatter.date(from: dateString))
+    }
+
+    func date(fromString maybeDateString: String?) -> Date? {
```

Comment:
I simplified this function since it looks like in all places where we're using it, we already know that the format will be in seconds and not milliseconds

Reply:
for consistency maybe we should name is iso8601SecondsDate...

Reply:
this method does not make assumptions for format, and I think it's better to keep it that way, so that it's usable by any date formatter. 
Also, it matches the original signature pretty closely, which makes it a pretty neat extension - it does the same as the original method, but allows you to send in an optional

Reply:
the reason that the other one _is_ static is that it does make assumptions about format, so it doesn't really apply to just any instance of `dateFormatter`